### PR TITLE
Vulkan: Disable VK_EXT_SAMPLER_FILTER_MINMAX in GCN AMD

### PR DESCRIPTION
**Pull request by FernandoS27**
Another broken hardware feature in AMD.

### CODE REVIEW

File: src/video_core/vulkan_common/vulkan_device.cpp [hunk 1]
```diff
@@ -587,6 +587,13 @@ Device::Device(VkInstance instance_, vk::PhysicalDevice physical_, VkSurfaceKHR
             ext_extended_dynamic_state = false;
         }
     }
+    if (ext_sampler_filter_minmax) {
+        if (driver_id == VK_DRIVER_ID_AMD_PROPRIETARY ||
+            driver_id == VK_DRIVER_ID_AMD_OPEN_SOURCE || driver_id == VK_DRIVER_ID_MESA_RADV) {
+            // Disable ext_sampler_filter_minmax in GCN as it is broken.
```
Comment (Morph1984) [inline src/video_core/vulkan_common/vulkan_device.cpp:593]:
```suggestion
            // Disable ext_sampler_filter_minmax on AMD GCN4 and lower as it is broken.
```
GCN4 covers polaris and lower

File: src/video_core/vulkan_common/vulkan_device.cpp [hunk 2]
```diff
@@ -587,6 +587,13 @@ Device::Device(VkInstance instance_, vk::PhysicalDevice physical_, VkSurfaceKHR
             ext_extended_dynamic_state = false;
         }
     }
+    if (ext_sampler_filter_minmax) {
+        if (driver_id == VK_DRIVER_ID_AMD_PROPRIETARY ||
+            driver_id == VK_DRIVER_ID_AMD_OPEN_SOURCE || driver_id == VK_DRIVER_ID_MESA_RADV) {
```
Comment (ameerj) [inline src/video_core/vulkan_common/vulkan_device.cpp:592]: Following the style of the rest of the function
```suggestion
    const bool is_amd = driver_id == VK_DRIVER_ID_AMD_PROPRIETARY ||
                        driver_id == VK_DRIVER_ID_MESA_RADV ||
                        driver_id == VK_DRIVER_ID_AMD_OPEN_SOURCE;
    if (ext_sampler_filter_minmax && is_amd) {
```

File: src/video_core/vulkan_common/vulkan_device.cpp [hunk 3]
```diff
@@ -587,6 +587,13 @@ Device::Device(VkInstance instance_, vk::PhysicalDevice physical_, VkSurfaceKHR
...
+            // Disable ext_sampler_filter_minmax in GCN as it is broken.
+            ext_sampler_filter_minmax = is_float16_supported;
```
Comment (ameerj) [inline src/video_core/vulkan_common/vulkan_device.cpp:594]:
```suggestion
    if (!is_float16_supported) {
        // LOG_WARNING(..., "Blaclisting AMD GCN for ...");
        ext_sampler_filter_minmax = false;
    }
```
Let's us be more explicit that HW not supporting FP16 is the target of being blacklisted for this extension by, and allows us to log it.

File: src/video_core/vulkan_common/vulkan_device.cpp [hunk 4]
```diff
@@ -587,19 +587,33 @@ Device::Device(VkInstance instance_, vk::PhysicalDevice physical_, VkSurfaceKHR
...
-    present_queue = logical.GetQueue(present_family);
+    const bool is_amd = driver_id == VK_DRIVER_ID_AMD_PROPRIETARY ||
+                        driver_id == VK_DRIVER_ID_MESA_RADV ||
+                        driver_id == VK_DRIVER_ID_AMD_OPEN_SOURCE;
+    if (ext_sampler_filter_minmax && is_amd) {
+        // Disable ext_sampler_filter_minmax on AMD GCN4 and lower as it is broken.
+        if (!is_float16_supported) {
+            LOG_WARNING(
+                Render_Vulkan,
+                "Blaclisting AMD GCN4 and lower for VK_EXT_SAMPLER_FILTER_MINMAX_EXTENSION_NAME");
```
Comment (Morph1984) [inline src/video_core/vulkan_common/vulkan_device.cpp:599]:
```suggestion
                "Blacklisting AMD GCN4 and lower for VK_EXT_SAMPLER_FILTER_MINMAX_EXTENSION_NAME");
```

File: src/video_core/vulkan_common/vulkan_device.cpp [hunk 5]
```diff
@@ -587,19 +587,33 @@ Device::Device(VkInstance instance_, vk::PhysicalDevice physical_, VkSurfaceKHR
...
+    if (ext_sampler_filter_minmax && is_amd) {
+        // Disable ext_sampler_filter_minmax on AMD GCN4 and lower as it is broken.
+        if (!is_float16_supported) {
+            LOG_WARNING(
+                Render_Vulkan,
+                "Blacklisting AMD GCN4 and lower for VK_EXT_SAMPLER_FILTER_MINMAX_EXTENSION_NAME");
+            ext_sampler_filter_minmax = false;
+        }
+    }
 }
```
Comment (Morph1984) [inline src/video_core/vulkan_common/vulkan_device.cpp:603]: remove this }

**Review by Morph1984 — APPROVED** [review]

**Pull request merged**